code/datums/quirks/positive_quirks.dm

```diff
@@ -207,6 +207,87 @@
 	// We've either added or removed TRAIT_NIGHT_VISION before calling this proc. Just refresh the eyes.
 	eyes.refresh()
 
+/datum/quirk/bilingual
+	name = "Bilingual"
+	desc = "Over the years you've picked up an extra language!"
+	icon = FA_ICON_GLOBE
+	value = 0
+	gain_text = span_notice("Some of the words of the people around you certainly aren't common. Good thing you studied for this.")
+	lose_text = span_notice("You seem to have forgotten your second language.")
+	medical_record_text = "Patient speaks multiple languages. Showoff."
+	var/possible_languages = list(
+		/datum/language/aphasia,
+		/datum/language/beachbum,
+		/datum/language/calcic,
+		/datum/language/draconic,
+		/datum/language/drone,
+		/datum/language/moffic,
+		/datum/language/monkey,
+		/datum/language/mushroom,
+		/datum/language/narsie,
+		/datum/language/nekomimetic,
+		/datum/language/piratespeak,
+		/datum/language/shadowtongue,
+		/datum/language/slime,
+		/datum/language/sylvan,
+		/datum/language/terrum,
+		/datum/language/uncommon,
+		/datum/language/voltaic,
+		/datum/language/xenocommon
+	)
+	var/extra_language
+	mail_goodies = list(/obj/item/taperecorder, /obj/item/clothing/head/frenchberet, /obj/item/clothing/mask/fakemoustache/italian)
+
+/datum/quirk/bilingual/add(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	//prevents yourself from learning a language you already have
+	for(var/datum/language/spoken as anything in possible_languages)
+		if(human_holder.has_language(spoken))
+			possible_languages -= list(spoken)
+	extra_language = possible_languages[rand(1, length(possible_languages))]
+	human_holder.grant_language(extra_language)
```

Comment:
You need to assign a source for this language, i.e. "LANGUAGE_QUIRK"

Reply:
I'm not sure I follow, this is the same way the other language quirk does it.

Reply:
Nevermind, updating both, oomfie!